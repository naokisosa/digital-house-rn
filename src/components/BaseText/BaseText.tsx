import { StyleSheet, Text as NativeText, TextProps as NativeTextProps } from 'react-native';
import React from 'react';

interface Props extends NativeTextProps {}

const BaseText = ({ children, style, ...rest }: Props) => {
  return (
    <NativeText style={[styles.base, style]} {...rest}>
      {children}
    </NativeText>
  );
};

export default BaseText;

const styles = StyleSheet.create({
  base: {
    fontFamily: 'Avenir',
  },
});
