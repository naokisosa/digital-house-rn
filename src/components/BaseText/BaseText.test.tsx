import 'react-native';
import React from 'react';

import BaseText from './BaseText';
import renderer from 'react-test-renderer';

describe('Test BaseText component', () => {
  it('renders correctly', () => {
    const { toJSON } = renderer.create(<BaseText>Mock</BaseText>);
    expect(toJSON()).toMatchSnapshot();
  });
});
