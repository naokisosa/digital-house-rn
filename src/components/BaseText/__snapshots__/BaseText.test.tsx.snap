// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BaseText component renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
        "fontFamily": "Avenir",
      },
      undefined,
    ]
  }
>
  Mock
</Text>
`;
